components/Participant.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { createRef, useEffect, useState } from 'react';
+
+const Participant = ({ gain, participant, isYou }) => {
+  const [videoRef] = useState(createRef<HTMLVideoElement>());
+
+  const playTrack = track => {
+    switch (track.kind) {
+      case 'audio':
+        if (!isYou) {
+          // Don't play your own audio
+          var audio = new Audio();
+          audio.srcObject = postProcessAudioTrack(track);
+        }
+        break;
+      case 'video':
+        taggedLog('Creating new MediaStream');
+        taggedLog('track:', track);
+        const { mediaStreamTrack } = track;
+        const stream = new MediaStream();
+        stream.addTrack(mediaStreamTrack);
+        videoRef.current!.srcObject = stream;
+        break;
+      default:
+        taggedLog('non-audio or video track:', track);
+    }
+  };
+
+  useEffect(() => {
+    participant.tracks.forEach(({ track }) => {
+      // sometimes track is `null` (?!), so check if it's real
+      track && playTrack(track);
+    });
+
+    participant.on('trackSubscribed', track => {
+      taggedLog(`trackSubscribed (${track.kind})`, track);
+      playTrack(track);
+    });
+  }, [participant, videoRef.current]);
+
+  const postProcessAudioTrack = audioTrack => {
+    // Manually add the stream so that we can post-process the audio
+    // @ts-ignore
+    const audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+    const stream = new MediaStream();
+    stream.addTrack(audioTrack.mediaStreamTrack);
+    const source = audioCtx.createMediaStreamSource(stream);
+
+    // mute
+    const gainNode = audioCtx.createGain();
+    gainNode.gain.setValueAtTime(gain, audioCtx.currentTime);
+
+    // hook it all up: source -> gain -> speakers
+    source.connect(gainNode).connect(audioCtx.destination);
```

Comment:
When/if this gets refactored to r-audio, all of this will be a nice lump of jsx, with a not-nice lump of dynamically set variables.